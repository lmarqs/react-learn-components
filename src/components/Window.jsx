import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import IsNightContext from '../contexts/IsNightContext'

const Window = ({ closed }) => {
  const isNight = useContext(IsNightContext);
  let classState = "window";
  if(closed)
    classState += " closed";

  if(isNight)
    classState += " light-on"

  return <div className={ classState }></div>
}
Window.propTypes = {
  closed: PropTypes.bool,
}

export default Window;
