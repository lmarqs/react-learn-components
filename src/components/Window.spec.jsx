import React from 'react';
import renderer from 'react-test-renderer';
import Window from './Window';

it('renders correctly without parameters', () => {
  const component = renderer.create(<Window />);

  expect(component.toJSON()).toMatchSnapshot();
});

it('renders correctly when "closed" is false', () => {
  const component = renderer.create(<Window closed={false} />);

  expect(component.toJSON()).toMatchSnapshot();
});

it('renders correctly when "closed" is true', () => {
  const component = renderer.create(<Window closed={true} />);

  expect(component.toJSON()).toMatchSnapshot();
});