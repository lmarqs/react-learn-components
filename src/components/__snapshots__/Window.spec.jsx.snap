// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when "closed" is false 1`] = `
<div
  className="window"
/>
`;

exports[`renders correctly when "closed" is true 1`] = `
<div
  className="window closed"
/>
`;

exports[`renders correctly without parameters 1`] = `
<div
  className="window"
/>
`;
